cypress/integration/chi-custom-elements/progress.spec.js

```diff
@@ -0,0 +1,28 @@
+describe('Progress', () => {
+  const selectors = ['progress-50-default', 'progress-50-danger'];
+
+  before(() => {
+    cy.visit('tests/custom-elements/progress.html');
+  });
+
+  beforeEach(() => {
+    selectors.forEach((s) => {
+      cy.get(`[data-cy=${s}]`, { timeout: 5000 })
+        .should('have.class', 'hydrated')
+        .children()
+        .should('match', 'progress')
+        .as(s);
+    });
+  });
+
+  it('Progress should be included inside the custom element', () => {
+    cy.get('@progress-50-default')
+      .should('match', 'progress');
+  });
+
+  it('Progress state should have appropriate class for state', () => {
```

Comment:
Could you explain me this test, please?

Reply:
@SergioQCL I'm checking whether the state attribute is setting correct class or not.